support spray-json

danlaudk opened the pull request:
i can create an example using akka-http that has both play-json and spray-json if acceptable

jbwheatley commented:
thanks @danlaudk - could you also update the readme here https://github.com/jbwheatley/pact4s#using-json-bodies

jbwheatley commented on build.sbt:
```diff
@@ -75,6 +75,16 @@ lazy val playJson =
+      libraryDependencies ++= Dependencies.sprayJson,
+      Test / parallelExecution := true
+    )
+    .dependsOn(shared)
```
**issue:** this should depend on `models`, not `shared`

<sub>This comment follows the [**conventionalcomments.org**](https://conventionalcomments.org/) standard</sub>

jbwheatley commented on spray-json/src/test/scala/pact4s/sprayjson/ImplicitsTests.scala:
```diff
@@ -0,0 +1,82 @@
+
+import munit.FunSuite
+import pact4s.sprayjson.implicits.providerStateFormat
+//import spray.json.DefaultJsonProtocol._
```
**nitpick:** this can be removed

<sub>This comment follows the [**conventionalcomments.org**](https://conventionalcomments.org/) standard</sub>

danlaudk commented:
hope i addressed the three comments, big and small (but important 😃). i removed a commit that was a merge of main into this branch. Then I rebased. So the state of the commits originally is only visible here on a copied branch https://github.com/jbwheatley/pact4s/compare/main...danlaudk:pact4s:spray-json-ok-not-squashed
